source/features/show-names.tsx

```diff
@@ -1,20 +1,42 @@
 import './show-names.css';
 import React from 'dom-chef';
 import select from 'select-dom';
+import elementReady from 'element-ready';
 import * as api from '../libs/api';
 import features from '../libs/features';
 import {getUsername} from '../libs/utils';
 
+const observer = new MutationObserver(async ([{addedNodes}]) => {
+	await usernames();
+
+	// Observe the new ajaxed-in containers
+	for (const node of addedNodes) {
+		if (node instanceof HTMLDivElement) {
+			observer.observe(node, {childList: true});
+		}
+	}
+});
+
 async function init(): Promise<false | void> {
-	// `a` selector needed to skip commits by non-GitHub users
-	const usernameElements = select.all('.js-discussion a.author:not(.rgh-fullname):not([href*="/apps/"]):not([href*="/marketplace/"]):not([data-hovercard-type="organization"])');
+	if (features.isDashboard()) {
+		observer.observe((await elementReady('#dashboard .news'))!, {childList: true});
```

Comment:
`onNewComments` waits for `domReady` and `elementReady` stops checking after `domReady`, i.e. this works exactly like a regular `select`

Reply:
Also you can probably use `observeEl` like:

```js
async updateAndWatch([{addedNodes}], observer) {
	await usernames();

	// Observe the new ajaxed-in containers
	for (const node of addedNodes) {
		if (node instanceof HTMLDivElement) {
			observer.observe(node, {childList: true});
		}
	}
}

async function init(): Promise<false | void> {
	if (features.isDashboard()) {
		observeEl('.news', updateAndWatch)
	} else {
		await usernames(); // This is called by observeEl, so it needs to be inside `else`
	}
}
```

Reply:
Running above (slightly altered) code results in an error, which doesn't makes sense to me:

![image](https://user-images.githubusercontent.com/55841/61189187-5c8db800-a68a-11e9-9136-7b12dc516585.png)

Reply:
See https://github.com/sindresorhus/refined-github/pull/2229/commits/0e1288059512f1db11b4d3e6e799b2ad4c957d56

Reply:
Indeed, that's an ugly error. It's because the function parameter is being destructured as `[{addedNodes}]` but `observeEl` only calls that callback with `[]`, so it fails when it tries to destructure `parameter[0]` (`undefined`) into `{addedNodes}`

Fix:

```ts
async function updateAndWatch([firstChange]: MutationRecord[], observer: MutationObserver): Promise<void> {
...
if (firstChange) {
	// use firstChange.addedNodes
```

Reply:
Implemented. Thanks.